### ISSUE TITLE
Disambiguate TargetFrameworkVersion and TargetFamework in help

### ISSUE THREAD

**Issue by am11** [Microsoft/msbuild#1805]
(moved from dotnet/cli#5921)

The command line help suggests:

```cmd
:: v1.6 is .NET Standard TFM
dotnet pack /p:TargetFrameworkVersion=v1.6 .
```

should work, however it errors out. Whereas `TargetFramework` switch does work:

```cmd
dotnet pack /p:TargetFramework=v1.6 .
```

The ask is to clarify the difference in help (perhaps with a core-specific example):

```cmd
dotnet msbuild -h
[snip]
        MSBuild MyApp.csproj /t:Clean
                             /p:Configuration=Debug;TargetFrameworkVersion=v3.5
::                                                   ^^^^^^^^^^  this one
```

**Comment by rainersigwald** [comment]
Oh, that _is_ confusing: one of the command-line examples given in [MSBuild's help](https://github.com/Microsoft/msbuild/blob/af0587665c448143579792638afa55f3499d6993/src/MSBuild/Resources/Strings.resx#L383) uses `TargetFrameworkVersion`, which was fine for non-Core Sdk projects building only for versions of the full framework, but is definitely confusing in the new world (and when passed out through `dotnet msbuild -h`, for example).

I think the best thing to do is just not use TargetFramework{anything} in the example. It's useful to specify two properties to show the syntax for that, though. Maybe go fully generic and specify something like `EnableFeatureX=true`?

**Comment by am11** [comment]
@rainersigwald, I agree that it should probably be a neutral property such as `EnableFeatureX=true` or perhaps even more templated form:

```
/p:Configuration=Debug;<key1>=<value1>;<key2>=<value2>
```

I tried to find some project-type-neutral property besides `Configuration` at https://msdn.microsoft.com/en-us/library/bb629394.aspx, but AFAIK -- as opposed what is documented there, `OutputPath` has no affect in `vcxproj`, whereas `OutDir` works there. Similarly `OutDir` is privately set in case of typical `csproj` and has no affect if we set it later (unless we re-invoke all the targets etc. - jeffkl/MSBuild-NetCore#2). So it's a no go. :(
